lib/fizzy/saas/authorization.rb

```diff
@@ -0,0 +1,28 @@
+module Fizzy
+  module Saas
+    module Authorization
+      module Controller
+        extend ActiveSupport::Concern
+
+        included do
+          before_action :ensure_only_employees_can_access_non_production_remote_environments, if: :authenticated?
+        end
+
+        private
+          def ensure_only_employees_can_access_non_production_remote_environments
+            head :forbidden if Rails.env.staging? && !Current.identity.employee?
+          end
+      end
+
+      module Identity
+        extend ActiveSupport::Concern
+
+        EMPLOYEE_DOMAINS = ["@37signals.com", "@basecamp.com"].freeze
```

Comment:
Can do 37signals.com only. We've been moving away from basecamp.com for staff email for ages and should stop accepting it in new places.

Reply:
https://3.basecamp.com/2914079/buckets/22311406/messages/9379506569